Lib/asyncio/timeouts.py

```diff
@@ -109,10 +109,16 @@ async def __aexit__(
         if self._state is _State.EXPIRING:
             self._state = _State.EXPIRED
 
-            if self._task.uncancel() <= self._cancelling and exc_type is exceptions.CancelledError:
+            if self._task.uncancel() <= self._cancelling:
                 # Since there are no new cancel requests, we're
                 # handling this.
-                raise TimeoutError from exc_val
+                if exc_type is exceptions.CancelledError:
+                    raise TimeoutError from exc_val
```

Comment:
What does this case mean? That the cancellation was not because of the timeout?

Reply:
Ah I get it, it's just a shortcut when no iteration is needed. OK.

Reply:
It is not a shortcut. It is a way to add TimeoutError at the top of exceptions chain. It is the current code.